### PR TITLE
Add mice infestation station trait

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This changes mice spawns to be a negative station trait instead of **always** appearing every round.  In the event that the mice trait does not get selected, it is still possible for mice to spawn as a random event.

## Why It's Good For The Game

Having to fix wires **EVERY** shift is tedious, especially on lowpop.  Some maps are worse than others with this, and it can lead to repeated power failures.  I think moving mice to a negative station trait is the correct solution to this problem, since it will not always be guaranteed. 

## Changelog

:cl:
balance: Mice maintenance infestations are now a negative station trait that is not guaranteed to occur.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
I am not convinced that mice in maint should be a station trait. When you talk about bad maps do you mean tram and how it clumps mice spawn in places like the engine room?

**Comment by Pickle-Coding** [comment]
Mice gives engineers something to do after the engine is set up and before other minor or major engineering problems occur. Mice can sometimes interfere with the emitters, which adds an extra task to set up the engine, making it less repetitive.

I think Lemon had a better idea on discord where engineers could choose to put an extreme amount of power in grid to allow plasma arcs to transfer energy to nearby yet disconnected cables.

**Comment by timothymtorres** [comment]
> I am not convinced that mice in maint should be a station trait. When you talk about bad maps do you mean tram and how it clumps mice spawn in places like the engine room?

Tram is the main culprit but this is a big problem with low pop due to a lack of engineers and maint access.

> Mice gives engineers something to do after the engine is set up and before other minor or major engineering problems occur. Mice can sometimes interfere with the emitters, which adds an extra task to set up the engine, making it less repetitive.

The argument isn't about _removing_ mice, since I agree with all your above points.  My changes affects their spawn frequency so it's not always 100% guaranteed.  The station trait for mice spawning is also weighted to be high since we don't want it to be a rare event.

TLDR; Mice spawning every round is a bad thing.  It's super tedious on lowpop but for highpop it isn't much of a problem since there are plenty of engineers.

**Comment by wesoda25** [comment]
I like that there are always many mice to be found, they're good for gimmicks.

**Comment by VladinXXV** [comment]
Maintenance sect kinda relies entirely on the existence of mice in maints. As does the regal pipegun, which already depends on a rare recipe book spawn and other componenets, maintenance drugs in general, which are hardly used as-is, and the occasional bounty here and there. Mice are a minor inconvenience and I think removing them ultimately does more harm than good because they tie into deceptively many playstyles and gimmicks. Even just as far as "that one lizard player who eats nothing but dead mice".

**Comment by nianjiilical** [comment]
ancetodtically i think the frequency of chewed wires could be lowered a *little* bit but i dont think it should be entirely shunted to a trait

but a trait that significantly ups the mice content and maybe makes regal rats more common isnt a bad idea imo

**Comment by NamelessFairy** [comment]
Very much not a fan of this, while minor it is job content for engineers with repairing broken cables. It can act as cover for antagonist engineers who want to disconnect parts of the station from the grid whether it be through just blaming the mice or if they're really clever planting burnt mice near cable breakages.

**Comment by Mothblocks** [comment]
I don't see the point either
